lib/utilities/validateQueryAndHeaders.js

```diff
@@ -17,13 +17,27 @@ function _validateKeys(unsupportedKeys, obj) {
 
 /**
  * validateQueryAndHeaders - Check request for unsupported queries or headers
- * @param {string} reqMethod - request method
- * @param {object} reqQuery - request query object
- * @param {object} reqHeaders - request headers object
+ * @param {object} request - request object
  * @param {object} log - Werelogs logger
  * @return {object} - empty object or object with error boolean property
  */
-function validateQueryAndHeaders(reqMethod, reqQuery, reqHeaders, log) {
+function validateQueryAndHeaders(request, log) {
+    const reqMethod = request.method;
+    const reqQuery = request.query;
+    const reqHeaders = request.headers;
+    const isBucketQuery = !request.objectKey;
+    // if the request is at bucket level, check for unsupported bucket queries
+    if (isBucketQuery) {
```

Comment:
I don't know the answer off the top of my head, but if we're distinguishing between bucket queries and not bucket queries, should any of the queries currently in the `UnsupportedQueries` array be in `UnsupportedBucketQueries` instead so this condition is hit?

Reply:
I think, ideally yes, because there are operations/queries that we support at object level and the same is not supported at bucket level, example `tagging`. Having said that I should reorganize the arrays.